frontends/web/src/routes/buy/utils.tsx

```diff
@@ -0,0 +1,46 @@
+/**
+ * Copyright 2022 Shift Crypto AG
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ExchangeDeals } from '../../api/exchanges';
+import { ExchangeProviderWithBestDeal, PaymentMethod } from './types';
+
+function findLowestFee(exchangeDeals: ExchangeDeals) {
+  let lowestFee = Object.values(exchangeDeals).map(providerDeals => Math.min(...providerDeals.map(deal => deal.fee)));
+  return Math.min(...lowestFee);
+}
+
+export function findBestDeal(exchangeDeals: ExchangeDeals): ExchangeProviderWithBestDeal[] {
+  // Get the lowest fee from all providers
+  const lowestFee = findLowestFee(exchangeDeals);
+
+  // Loop through each provider in the exchangeDeals object
+  let arr: ExchangeProviderWithBestDeal[] = [];
+  Object.keys(exchangeDeals).forEach((key) => {
+    let providerName = key as keyof ExchangeDeals ;
+    const providerObj = { providerName, methods: exchangeDeals[providerName] as PaymentMethod[] };
+    arr = [...arr, providerObj];
+  });
+
+  //returns array of object of all providers
+  //and each of its payment method has `isBestDeal` property
+  return arr.map(provider => ({
+    ...provider,
+    methods: provider.methods.map(method => ({
+      ...method,
+      isBestDeal: method.fee === lowestFee
```

Comment:
The `best deal` tag makes sense only if there is more than one option. You could add a check on the number of available providers (after filtering them upon region)

Reply:
Yup. makes sense. I'll add a check on the number of providers, but I'll keep `isBestDeal` property to exist regardless of the number of providers available (even if theres only 1 provider) to keep consistency. So `bestDeal` will always default to false when there's only one provider.